### ISSUE TITLE
Clean up resolutions in NPM dependencies

### ISSUE THREAD

**Issue by stefanobaghino-da** [digital-asset/daml#13383]
We frequently use Yarn's `resolutions` field in the `package.json` file to quickly fix security alerts in our packages. While this works, it might be good for us to clean up those where we can. If the issue are transitive dependencies not being upgrade along the way are the issue, pushing those changes to the supply chain is also in the scope for this ticket.

Time box this to one day. If there are still resolutions that need fixing, open a new ticket to work on it as part of a future gardening session.

**Comment by S11001001** [comment]
1. Check `yarn.lock` to see whether a constraint occurs only at the top-level, or is imposed under `dependencies`.
2. If the latter, remove from `resolutions` in `package.json`.
3. Run `yarn` in the directory.
4. Check `yarn.lock` again to ensure there were no version regressions.

**Comment by S11001001** [comment]
@stefanobaghino-da Is the above what you had in mind? Is there a command I can run locally to check for no regressions with regard to the security alerts?

**Comment by stefanobaghino-da** [comment]
Rather than checking the `yarn.lock` manually, I would recommend maybe:
1. fix any outstanding @dependabot alert before starting
2. remove resolutions one by one
3. for each one, run `yarn audit` and see if a new version can be used instead of using a resolution
4. repeat until the end of the time box
5. create a new ticket if there are outstanding resolutions after the process

I'm not 100% sure of the usefulness of this exercise to be honest but I find the resolutions to be a possible source of errors in the future if we upgrade a library and we start seeing cascading failures that are difficult to diagnose. I'll leave it up to the team to decide whether I'm simply being pedantic about pure form or raising a valid concern.
